.github/workflows/python-ci.yml

```diff
@@ -65,7 +65,7 @@ jobs:
     runs-on: macos-latest
     strategy:
       matrix:
-        python-version: [ "3.10.8" ]
+        python-version: [ "3.12.1" ]
```

Comment:
`macos-latest` jobs are running on M2 machines now, and there was no python 3.10.8 available for that architecture. I took advantage of that situation to bring us closer to the latest stable python version.